applications/nrf_desktop/src/modules/dfu.c

```diff
@@ -754,6 +860,13 @@ static bool app_event_handler(const struct app_event_header *aeh)
 			k_work_init_delayable(&background_erase, background_erase_handler);
 			k_work_init_delayable(&background_store, background_store_handler);
 
+			if (IS_ENABLED(CONFIG_DESKTOP_DFU_LOCK)) {
+				bool is_lock_claimed;
+
+				is_lock_claimed = dfu_lock_claim(&config_channel_owner);
```

Comment:
Maybe we could combine claiming lock with background erase scheduling for the firs time (both operations must be done together)